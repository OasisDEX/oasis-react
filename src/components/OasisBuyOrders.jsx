import React, { PureComponent } from 'react';
import { PropTypes } from 'prop-types';
// import ImmutablePropTypes from 'react-immutable-proptypes';

import OasisWidgetFrame from '../containers/OasisWidgetFrame';
import OasisTable from './OasisTable';
import { isOfferOwner, toDisplayFormat } from '../utils/orders';

const propTypes = PropTypes && {};
const defaultProps = {};

const actionsColumnTemplate = function(offer) {
  const onCancel = () => this.cancelOffer(offer);
  return (
    isOfferOwner(offer) ? (<button onClick={onCancel}>cancel</button>): null
  );
};

const colsDefinition = (baseToken, quoteToken, orderActions) => {
  return [
    { heading: `price`, key: 'bid_price' },
    { heading: `${quoteToken}`, key: 'sell_how_much' },
    { heading: `${baseToken}`, key: 'buy_how_much'},
    { heading: ``, template: actionsColumnTemplate.bind(orderActions)}
  ];
};


class OasisBuyOrders extends PureComponent {
  render() {
    const { activeTradingPair: { baseToken, quoteToken }, buyOffers = [], cancelOffer } = this.props;
    const orderActions = { cancelOffer };
    return (
      <OasisWidgetFrame heading={'BUY ORDERS'}>
        <OasisTable
          rows={buyOffers.sort((p, c) => p.bid_price_sort < c.bid_price_sort ? 1 : -1).map(toDisplayFormat)}
          col={colsDefinition(baseToken, quoteToken, orderActions)}
        />
      </OasisWidgetFrame>
    );
  }
}

OasisBuyOrders.displayName = 'OasisBuyOrders';
OasisBuyOrders.propTypes = propTypes;
OasisBuyOrders.defaultProps = defaultProps;
export default OasisBuyOrders;
