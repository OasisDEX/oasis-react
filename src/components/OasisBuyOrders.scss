@import "../styles/modules/constants";
@import "../styles/modules/grid";

.table {

  td, th {
    text-align: right;
  }

  td {
    font-weight: 500;
  }

  th:first-child,
  td:first-child {
    flex-basis: 170px;
    text-align: right;
  }

  td:first-child {
    color: $success-text;
  }

  th:nth-child(2),
  td:nth-child(2) {
    flex-basis: 110px;
    text-align: right;
  }

  th:nth-child(3),
  td:nth-child(3) {
    flex-basis: 110px;
    text-align: right;
  }

  th:last-child,
  td:last-child {
    @include reset;
  }


}
