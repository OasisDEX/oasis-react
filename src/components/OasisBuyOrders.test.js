/* global shallow describe it expect */
/* eslint-disable import/first */
import React from 'react';
import { shallow } from 'enzyme';
import OasisBuyOrders from './OasisBuyOrders';

describe('(Component) OasisBuyOrders', () => {
  it('should render', () => {
    const props = {

    };
    const wrapper = shallow(
      <OasisBuyOrders {...props}/>,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
