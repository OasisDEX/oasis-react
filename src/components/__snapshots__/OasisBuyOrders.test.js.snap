// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) OasisBuyOrders should render 1`] = `
<OasisWidgetFrame
  className=""
  heading="BUY ORDERS"
  loadProgressSection={
    <LoadProgressSection
      loadedOffersList={
        Immutable.List [
        ]
      }
      offersTotalCount={undefined}
    />
  }
  loadingDataText="buy orders"
>
  <OasisTable
    col={
      Array [
        Object {
          "heading": "price",
          "template": [Function],
        },
        Object {
          "heading": "W-ETH",
          "template": [Function],
        },
        Object {
          "heading": "MKR",
          "template": [Function],
        },
        Object {
          "heading": "",
          "template": [Function],
        },
      ]
    }
    collapseEnabled={false}
    collapseInitial={false}
    collapseRowNumber={4}
    onRowClick={[Function]}
    rows={
      Immutable.List [
      ]
    }
  />
</OasisWidgetFrame>
`;
